### PR TITLE
Added feature-gate to glossary

### LINKED ISSUE #38848: Add “feature gate” to glossary
**This is a Feature Request**

**What would you like to be added**
Let's explain what a feature gate is with an entry in our [glossary](https://kubernetes.io/docs/reference/glossary/).

**Why is this needed**
Lots of pages link to https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ mainly by way of explaining what the term means. A glossary entry could really help.

**Comments**
/kind feature
/language en

### PR DESCRIPTION
**Pull request by T-Lakshmi**
Added a new entry in glossary by explaining term feature gates.
Fixes #38848.

### CODE REVIEW

**Comment by T-Lakshmi** [comment]
@kbhawkey, @reylejano,
PTAL!!

**Comment by T-Lakshmi** [comment]
Preview page: https://deploy-preview-38895--kubernetes-io-main-staging.netlify.app/docs/reference/glossary/?fundamental=true#term-feature-gates

File: content/en/docs/reference/glossary/feature-gates.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+tags:
+- fundamental
+- operation
+---
+
+Feature gates are a set of key=value pairs that describe Kubernetes features.
+
+<!--more-->
+
+You can turn these features on or off using the --feature-gates command line flag on each Kubernetes component.
```
Comment (divya-mohan0209) [inline content/en/docs/reference/glossary/feature-gates.md:19]: nit.
```suggestion
You can turn these features on or off using the `--feature-gates` command line flag on each Kubernetes component.
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]

**Comment by T-Lakshmi** [comment]
@divya-mohan0209 ,
Thanks, updated as per your suggestion. PTAL!

File: content/en/docs/reference/glossary/feature-gates.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+title: Feature gates
+id: feature-gates
+date: 2023-01-12
+full_link: /docs/reference/command-line-tools-reference/feature-gates/
+short_description: >
+  Feature gates are a set of key=value pairs that describe Kubernetes features.
```
Comment (sftim) [inline content/en/docs/reference/glossary/feature-gates.md:7]:
```suggestion
  A way to control whether or not a particular Kubernetes feature is enabled.
```

File: content/en/docs/reference/glossary/feature-gates.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+title: Feature gates
+id: feature-gates
```
Comment (sftim) [inline content/en/docs/reference/glossary/feature-gates.md:3]:
```suggestion
title: Feature gate
id: feature-gate
```

File: content/en/docs/reference/glossary/feature-gates.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+short_description: >
+  Feature gates are a set of key=value pairs that describe Kubernetes features.
+
+aka: 
+tags:
+- fundamental
+- operation
+---
+
+Feature gates are a set of key=value pairs that describe Kubernetes features.
```
Comment (sftim) [inline content/en/docs/reference/glossary/feature-gates.md:15]:
```suggestion
Feature gates are a set of keys (opaque string values) that you can use to control which
Kubernetes features are enabled in your cluster.
```

File: content/en/docs/reference/glossary/feature-gates.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+<!--more-->
+
+You can turn these features on or off using the `--feature-gates` command line flag on each Kubernetes component.
+Each Kubernetes component lets you enable or disable a set of feature gates that are relevant to that component. Use -h flag to see a full set of feature gates for all components.
```
Comment (sftim) [inline content/en/docs/reference/glossary/feature-gates.md:20]:
```suggestion
Each Kubernetes component lets you enable or disable a set of feature gates that are relevant to that component.
The Kubernetes documentation lists all current
[feature gates](/docs/reference/command-line-tools-reference/feature-gates/) and what they control.
```

**Review by sftim — COMMENTED** [review]
What do you think of these suggestions?

File: content/en/docs/reference/glossary/feature-gates.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/reference/glossary/feature-gates.md:15]: The values (enabled or disabled) aren't part of the feature gate. For example, `IndexedJob` is a feature gate you can enable; `IndexedJob=true` isn't.

Comment (T-Lakshmi) [reply content/en/docs/reference/glossary/feature-gates.md]: @sftim, 
Thanks for the review comments. I incorparated the suggestions. PTAL!!

File: content/en/docs/reference/glossary/feature-gates.md (hunk 5 above)
Comment (T-Lakshmi) [reply content/en/docs/reference/glossary/feature-gates.md]: @sftim,
Thanks for the review comments. I incorparated the suggestions. PTAL!!

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm

**Comment by T-Lakshmi** [comment]
/assign @krol3
could you please take a look.

File: content/en/docs/reference/glossary/feature-gates.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+- fundamental
+- operation
+---
+
+Feature gates are a set of keys (opaque string values) that you can use to control which
+Kubernetes features are enabled in your cluster.
+
+<!--more-->
+
+You can turn these features on or off using the '--feature-gates' command line flag on each Kubernetes component.
```
Comment (divya-mohan0209) [inline content/en/docs/reference/glossary/feature-gates.md:20]: Just a small nit. I'll approve & lgtm it once this is fixed
```suggestion
You can turn these features on or off using the `--feature-gates` command line flag on each Kubernetes component.
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]

**Comment by T-Lakshmi** [comment]
@divya-mohan0209 ,
updated, Thanks.

**Review by divya-mohan0209 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**